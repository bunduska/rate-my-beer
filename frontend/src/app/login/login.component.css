.flex-container {
  background-color: gray;
  opacity: 0.8;
  width: 40%;
  margin-left: 30%;
  align-content: center;
  padding-bottom: 100px;
  border-top-width: 100px;
  border-radius: 20px;
  display: flex;
  align-items: center;
  flex-direction: column;
  flex-wrap: wrap;
  align-content: center;
}

.icon-display {
  transform: scale(0.7);
}

.h1-content {
  margin: 50px 0 25px 0;
  font-size: xx-large;
}

.h1-text {
  font-size: xxx-large;
  font-weight: 500;
}

mat-icon {
  margin: 0 10px 0 5px;
}

.mat-icon.mat-primary {
  color: white;
}

.mat-raised-button {
  width: 250px;
  height: 50px;
}

.register-button {
  margin-top: 50px;
}

.mat-flat-button.mat-primary, .mat-raised-button.mat-primary, .mat-fab.mat-primary, .mat-mini-fab.mat-primary {
  color: white;
}

.mat-button-wrapper {
  font-size: 20px;
}

.mat-form-field {
  display: flex;
  width: min(500px, 60vw);
  justify-content: center;
  outline-color: blueviolet;
}

.mat-error {
  color: blue;
  font-size: x-large;
  font-weight: bolder;
  padding-top: 20px;
}

.image {
  background-color: white;
  background-image: url("../../assets/beer-background-royalty-free-image-1608331890_.jpg");
  background-size: 100% 100%;
  background-repeat: no-repeat;
  background-position: left top;
  height: 100vh;
}