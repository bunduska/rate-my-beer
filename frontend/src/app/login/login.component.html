<div class="image">
    <form [formGroup]="loginForm" (ngSubmit)="onSubmit()">
        <div class="flex-container">
            <div class="h1-content">
                <h1><span class="h1-text">Login</span></h1>
            </div>
            <mat-form-field appearance="outline">
                <mat-label>Email</mat-label>
                <mat-icon color="primary" matPrefix>email</mat-icon>
                <input class="inputClass" matInput placeholder="ticket@example.com" formControlName="email"
                    type="email" />
            </mat-form-field>
            <mat-error *ngIf="submitted && f.email.errors">Email is required</mat-error>
            <mat-form-field appearance="outline">
                <mat-label>Password</mat-label>
                <mat-icon matPrefix color="primary" class="icon-display"></mat-icon>

                <input class="inputClass" matInput [type]="hide ? 'password' : 'text'" formControlName="password" />
                <a mat-icon-button matSuffix (click)="hide = !hide" [attr.aria-label]="'Hide password'"
                    [attr.aria-pressed]="hide" class="hide">
                    <mat-icon color="primary">{{
                        hide ? 'visibility_off' : 'visibility'
                        }}</mat-icon>
                </a>
            </mat-form-field>
            <mat-error *ngIf="submitted && f.password.errors">
                <div *ngIf="f.password.errors.required">Password is required</div>
                <div *ngIf="f.password.errors.minlength">
                    Password must be at least 8 characters
                </div>
            </mat-error>
            <div class="register-button">
                <button mat-raised-button color="primary">
                    Login
                </button>
            </div>
            <br />
            <div class="button">
                <button mat-raised-button routerLink="/register">Or Register</button>
            </div>
        </div>
    </form>
</div>