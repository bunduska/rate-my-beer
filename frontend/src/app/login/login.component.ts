import { Component, OnInit } from '@angular/core';
import { Router } from '@angular/router';
import { FormBuilder, FormGroup, Validators } from '@angular/forms';
import { first } from 'rxjs/operators';
import { AuthService } from '../services/auth.service';
import { MatSnackBar } from '@angular/material/snack-bar';

@Component({
  selector: 'app-login',
  templateUrl: './login.component.html',
  styleUrls: ['./login.component.css'],
})
export class LoginComponent implements OnInit {
  loginForm!: FormGroup;
  submitted: boolean = false;
  wrongInputMessage: string = '';
  currentUser: any;
  hide: boolean = true;

  constructor(
    private formBuilder: FormBuilder,
    private router: Router,
    private authService: AuthService,
    private snackBar: MatSnackBar,
  ) {
    this.currentUser = this.authService.getCurrentUser();
    if (this.currentUser !== null) {
      this.router.navigate(['/']);
    }
  }

  ngOnInit(): void {
    this.loginForm = this.formBuilder.group({
      email: ['', Validators.required],
      password: ['', Validators.required],
    });
  }

  get f() {
    return this.loginForm.controls;
  }

  onSubmit(): void {
    this.submitted = true;

    if (this.loginForm.invalid) {
      return;
    }

    this.authService
      .login(this.f.email.value, this.f.password.value)
      .pipe(first())
      .subscribe((data) => {
        if (data.message !== undefined) {
          this.wrongInputMessage = data.message;
          this.openSnackBar(this.wrongInputMessage);
        } else {
          this.router.navigate(['/']);
        }
      });
  }

  openSnackBar(message: string) {
    this.snackBar.open(this.wrongInputMessage, 'OK', { duration: 4000 });
  }
}
