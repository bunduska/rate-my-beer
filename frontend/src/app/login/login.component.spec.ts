import { ComponentFixture, TestBed } from '@angular/core/testing';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { LoginComponent } from './login.component';
import { MatInputModule } from '@angular/material/input';
import { MatIconModule } from '@angular/material/icon';
import { RouterTestingModule } from '@angular/router/testing';
import { HttpClientModule } from '@angular/common/http';
import { MatSnackBar } from '@angular/material/snack-bar';
import { OverlayModule } from '@angular/cdk/overlay';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

describe('LoginComponent', () => {
  let component: LoginComponent;
  let fixture: ComponentFixture<LoginComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [LoginComponent],
      imports: [
        FormsModule,
        ReactiveFormsModule,
        BrowserAnimationsModule,
        OverlayModule,
        MatInputModule,
        MatIconModule,
        RouterTestingModule,
        HttpClientModule,
      ],
      providers: [MatSnackBar],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(LoginComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match with previous snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
