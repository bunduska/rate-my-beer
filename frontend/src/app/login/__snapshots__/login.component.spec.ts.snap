// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent should match with previous snapshot 1`] = `
<app-login
  authService={[Function AuthService]}
  currentUser="null"
  formBuilder={[Function FormBuilder]}
  hide={[Function Boolean]}
  loginForm={[Function FormGroup]}
  router={[Function Router]}
  snackBar={[Function MatSnackBar]}
  submitted="false"
  wrongInputMessage=""
>
  <div
    class="image"
  >
    <form
      class="ng-untouched ng-pristine ng-invalid"
      novalidate=""
    >
      <div
        class="flex-container"
      >
        <div
          class="h1-content"
        >
          <h1>
            <span
              class="h1-text"
            >
              Login
            </span>
          </h1>
        </div>
        <mat-form-field
          appearance="fill"
          class="mat-form-field mat-primary mat-form-field-type-mat-input mat-form-field-appearance-fill mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder ng-untouched ng-pristine ng-invalid"
        >
          <div
            class="mat-form-field-wrapper"
          >
            <div
              class="mat-form-field-flex"
            >
              <div
                class="mat-form-field-prefix"
              >
                <mat-icon
                  aria-hidden="true"
                  class="mat-icon notranslate mat-primary material-icons"
                  color="primary"
                  data-mat-icon-type="font"
                  matprefix=""
                  role="img"
                >
                  email
                </mat-icon>
              </div>
              <div
                class="mat-form-field-infix"
              >
                <input
                  aria-invalid="false"
                  aria-required="false"
                  class="inputClass mat-input-element mat-form-field-autofill-control ng-untouched ng-pristine ng-invalid cdk-text-field-autofill-monitored"
                  data-placeholder="beer@drinking.com"
                  formcontrolname="email"
                  matinput=""
                  type="email"
                />
                <span
                  class="mat-form-field-label-wrapper"
                >
                  <label
                    class="mat-form-field-label mat-empty mat-form-field-empty"
                  >
                    <mat-label>
                      Email
                    </mat-label>
                  </label>
                </span>
              </div>
            </div>
            <div
              class="mat-form-field-underline"
            >
              <span
                class="mat-form-field-ripple"
              />
            </div>
            <div
              class="mat-form-field-subscript-wrapper"
            >
              <div
                class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
                style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
              >
                <div
                  class="mat-form-field-hint-spacer"
                />
              </div>
            </div>
          </div>
        </mat-form-field>
        <mat-form-field
          appearance="fill"
          class="mat-form-field mat-primary mat-form-field-type-mat-input mat-form-field-appearance-fill mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder ng-untouched ng-pristine ng-invalid"
        >
          <div
            class="mat-form-field-wrapper"
          >
            <div
              class="mat-form-field-flex"
            >
              <div
                class="mat-form-field-prefix"
              >
                <mat-icon
                  aria-hidden="true"
                  class="icon-display mat-icon notranslate mat-primary material-icons"
                  color="primary"
                  data-mat-icon-type="font"
                  matprefix=""
                  role="img"
                />
              </div>
              <div
                class="mat-form-field-infix"
              >
                <input
                  aria-invalid="false"
                  aria-required="false"
                  class="inputClass mat-input-element mat-form-field-autofill-control ng-untouched ng-pristine ng-invalid cdk-text-field-autofill-monitored"
                  formcontrolname="password"
                  matinput=""
                  type="password"
                />
                <span
                  class="mat-form-field-label-wrapper"
                >
                  <label
                    class="mat-form-field-label mat-empty mat-form-field-empty"
                  >
                    <mat-label>
                      Password
                    </mat-label>
                  </label>
                </span>
              </div>
              <div
                class="mat-form-field-suffix"
              >
                <a
                  aria-label="Hide password"
                  aria-pressed="true"
                  class="hide"
                  color="primary"
                  mat-icon-button=""
                  matsuffix=""
                >
                  <mat-icon
                    aria-hidden="true"
                    class="mat-icon notranslate material-icons mat-icon-no-color"
                    data-mat-icon-type="font"
                    role="img"
                  >
                    visibility_off
                  </mat-icon>
                </a>
              </div>
            </div>
            <div
              class="mat-form-field-underline"
            >
              <span
                class="mat-form-field-ripple"
              />
            </div>
            <div
              class="mat-form-field-subscript-wrapper"
            >
              <div
                class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
                style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
              >
                <div
                  class="mat-form-field-hint-spacer"
                />
              </div>
            </div>
          </div>
        </mat-form-field>
        <div
          class="register-button"
        >
          <button
            color="accent"
            mat-raised-button=""
          >
             Login 
          </button>
        </div>
        <br />
        <div
          class="button"
        >
          <button
            color="primary"
            mat-raised-button=""
            routerlink="/register"
            tabindex="0"
          >
            Or Register
          </button>
        </div>
      </div>
    </form>
  </div>
</app-login>
`;
